packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/multiphysics/DecapodesContext.java

```diff
@@ -0,0 +1,17 @@
+package software.uncharted.terarium.hmiserver.models.dataservice.multiphysics;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import java.io.Serializable;
+import lombok.Data;
+import lombok.experimental.Accessors;
+
+import software.uncharted.terarium.hmiserver.annotations.TSModel;
+import software.uncharted.terarium.hmiserver.models.dataservice.TerariumAsset;
+
+@Data
+@Accessors(chain = true)
+@TSModel
+public class DecapodesContext extends TerariumAsset implements Serializable {
+	private ContextHeader header;
```

Comment:
Because you are using `TerariumAsset` you do not need a `ContextHeader`.
`TerariumAsset` provides an `id, name, description`, only the `parentModel` is missing, that's why I'm using the `ModelHeader` in ticket #2638
```java
private ModelHeader header;
private Object context;
@JsonProperty("parent_model")
private UUID parentModel;
```

Reply:
It needs to be `header.parent_model`, so as far as I am aware, without using a DSL and a generator like intertypes to do selective unions, we would either need to 
- extend ModelHeader and have extra fields we dont need, e.g. "extracted_from", or
- create new class with exactly the field that were prescribed

I went with the 2nd option.

Reply:
I think if we want to have extra fields that may be okay, but if want to change the data structure that can be quite annoying as there may be schema-validity checkers downstream, so either way we need to adhere to the agreed upon schema.

Reply:
Okay, I personally prefer to have the controller do that job, but the model will do.